torchsde/_brownian/brownian_interval.py

```diff
@@ -467,12 +468,15 @@ def increment_and_space_time_levy_area_cache(interval):
     def _increment_and_space_time_levy_area(self):
         return self._w_h
 
-    def __call__(self, ta, tb):
+    # TODO: pick better names for return_U, return_A. A is clearly 'levy_area', but U?
+    def __call__(self, ta, tb=None, return_U=False, return_A=False):
+        if tb is None:
+            ta, tb = self.start, ta
         ta = float(ta)
         tb = float(tb)
         # Can get queries just inside and outside the specified region in SDE solvers; we just clamp.
-        ta = min(self.start, max(ta, self.end))
-        tb = min(self.start, max(tb, self.end))
+        ta = max(self.start, min(ta, self.end))
```

Comment:
This seems like it will fail silently when `ta` or `tb` is out of range. Typically, it shouldn't be a big problem, since we don't expect values that are out of range by too much.

But could we at least raise a warning or something like that, just to defend against the imaginary adversarial user?

Reply:
That would raise a warning as a result of normal solver behaviour as well, though.

FWIW I'm inclined to actually regard this as bug in the solvers. We don't have to step past endpoint and interpolate back: we can just truncate our final step to exactly match the endpoint.

Then this line can be switched for an error.

Reply:
> FWIW I'm inclined to actually regard this as bug in the solvers. We don't have to step past endpoint and interpolate back: we can just truncate our final step to exactly match the endpoint. Then this line can be switched for an error.

I agree with this. 

Overall, I think the expected behavior would be something like this
```python
bm = BrownianInterval(t0=0., t1=1.0, ...)
y0 = torch.zeros(3, 4)
ts = torch.tensor([0., 2.])
sdeint(y0, ts, bm=bm)  # Raises warning/error eventually
```

What do you think? Would you like to give it a go?

Reply:
Can do. I'll do it as a separate PR though, so that you can work off of this one in your branches.

Reply:
(Hah, you ninja'd my comment with a merge anyway.)

Reply:
Sorry, was trying to move fast. Thought you'd do it separately. Anyway, I'll be more careful next time!